### PR TITLE
backport!: Merge bitcoin#19915, 19469, 19464, 19512, 22495

### PR DESCRIPTION
**Pull request by vijaydasmp**
bitcoin backports

### CODE REVIEW

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 In this PR I tried to backport changes related to deprecation of banscore from following [bitcoin#19464](https://github.com/bitcoin/bitcoin/pull/19464), [bitcoin#19469](https://github.com/bitcoin/bitcoin/pull/19469), [bitcoin#19512](https://github.com/bitcoin/bitcoin/pull/19512) , requesting suggestion for updating test/functional/p2p_quorum_data.py to handle banscore deprecation

**Comment by vijaydasmp** [comment]
Hello @knst , In this [PR ](https://github.com/dashpay/dash/pull/5511)I am backporting banscore deprecation related commits,  [19512](https://github.com/bitcoin/bitcoin/pull/19512), [19464](https://github.com/bitcoin/bitcoin/pull/19464), [19469](https://github.com/bitcoin/bitcoin/pull/19469),  In addition to backported changes, there is one more test/functional/p2p_quorum_data.py which needs to updated.
requesting change suggestion in testcase p2p_quorum_data.py

**Comment by PastaPastaPasta** [comment]
Note that this includes breaking changes and as such cannot be merged until we are merging for v21; currently merging for v20.1

File: doc/release-notes-19464.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Updated settings
+----------------
+
+- The `-banscore` configuration option, which modified the default threshold for
+  disconnecting and discouraging misbehaving peers, has been removed as part of
+  changes in this release to the handling of misbehaving peers. Refer to the
+  section, "Changes regarding misbehaving peers", for details. (#19464)
```
Comment (UdjinM6) [inline doc/release-notes-19464.md:7]: nit:
```suggestion
  section, "Changes regarding misbehaving peers", for details. (dash#5511)
```

File: doc/release-notes-19469.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Updated RPCs
+------------
+
+- `getpeerinfo` no longer returns the `banscore` field unless the configuration
+  option `-deprecatedrpc=banscore` is used. The `banscore` field will be fully
+  removed in the next major release. (#19469)
```
Comment (UdjinM6) [inline doc/release-notes-19469.md:6]:
```suggestion
  removed in the next major release. (dash#5511)
```

**Review by UdjinM6 — COMMENTED** [review]
2 small suggestions. CI failure is unrelated and should be fixed via #5932

File: doc/release-notes-19469.md (hunk 2 above)
Comment (vijaydasmp) [reply doc/release-notes-19469.md:6]: fixed

Comment (UdjinM6) [reply doc/release-notes-19469.md:6]: nope

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta , requesting review

File: src/rpc/net.cpp [hunk 3]
```diff
@@ -224,7 +224,10 @@ static UniValue getpeerinfo(const JSONRPCRequest& request)
         obj.pushKV("masternode", stats.m_masternode_connection);
         obj.pushKV("startingheight", stats.nStartingHeight);
         if (fStateStats) {
-            obj.pushKV("banscore", statestats.m_misbehavior_score);
+            if (IsDeprecatedRPCEnabled("banscore")) {
+                // banscore is deprecated in v0.21 for removal in v0.22
```
Comment (knst) [inline src/rpc/net.cpp:228]: Dash uses other version numberation for v21, v22

```suggestion
                // banscore is deprecated in v21 for removal in v22
```

**Review by knst — COMMENTED** [review]
please check comments.

otherwise LGTM

File: src/rpc/net.cpp (hunk 3 above)
Comment (vijaydasmp) [reply src/rpc/net.cpp:228]: Implemented Suggestion

**Review by knst — APPROVED** [review]
utACK e9157563290c178eeeef1cb92830906545fc970a

File: src/qt/forms/debugwindow.ui [hunk 4]
```diff
@@ -1189,37 +1189,14 @@
...
-                 <property name="textFormat">
-                  <enum>Qt::PlainText</enum>
-                 </property>
-                 <property name="textInteractionFlags">
-                  <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
-                 </property>
-                </widget>
-               </item>
-               <item row="12" column="0">
+               <item row="10" column="0">
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply src/qt/forms/debugwindow.ui]: Good catch! :+1:

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta , requesting review

**Comment by vijaydasmp** [comment]
Due to conflict after review done, I will have to rebase to fix this , requesting review

**Comment by UdjinM6** [comment]
Looks good but should be merged after #5939 to avoid issues

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @knst @PastaPastaPasta, requesting review

File: src/rpc/net.cpp (hunk 3 above)
Comment (PastaPastaPasta) [reply src/rpc/net.cpp]: you did this in the wrong commit; but I think it's fine enought

**Review by PastaPastaPasta — APPROVED** [review]
utACK a39e85dc1cde18b6f5805bb3aee68d12326528ff

**Comment by PastaPastaPasta** [comment]
conflicts with #5933

**Comment by PastaPastaPasta** [comment]
I rebased it for you. Range diff here:
```
 -:  ---------- >  1:  10fa7a66b6 Merge #19538: ci: Add tsan suppression for race in DatabaseBatch
 -:  ---------- >  2:  2489f29f0e Merge #19830: test: Add tsan supp for leveldb::DBImpl::DeleteObsoleteFiles
 -:  ---------- >  3:  59d5a4ef39 Merge #19773: wallet: Avoid recursive lock in IsTrusted
 -:  ---------- >  4:  5821a1d23a Merge #14582: wallet: always do avoid partial spends if fees are within a specified range
 -:  ---------- >  5:  6a6d379711 Merge #19756: tests: add sync_all to fix race condition in wallet groups test
 -:  ---------- >  6:  eb4270deae Merge #19743: -maxapsfee follow-up
 -:  ---------- >  7:  1707f01309 fix: follow-up changes from bitcoin/bitcoin#22220 for maxapsfee
 -:  ---------- >  8:  44929bad82 Merge #16404: qa: Test ZMQ notification after chain reorg
 -:  ---------- >  9:  a1c2386153 Merge #17445: zmq: Fix due to invalid argument and multiple notifiers
 -:  ---------- > 10:  0ce66fd477 Merge #19507: Expand functional zmq transaction tests
 -:  ---------- > 11:  2f788aa76d fix: change port to use for zmq in interface_zmq_dash.py
 -:  ---------- > 12:  5a6b8b6b1f partial Merge bitcoin/bitcoin#27053: wallet: reuse change dest when re-creating TX with avoidpartialspends
 -:  ---------- > 13:  2dd6082e54 refactor: move special transaction processing into helper class
 -:  ---------- > 14:  0d6f736a19 fix: add missing entity for destruction in DashTestSetupClose
 -:  ---------- > 15:  667852c851 refactor: cleanup CDSNotificationInterface member names, add asserts
 -:  ---------- > 16:  67cfee70f8 refactor: remove CMNHFManager::GetSignalsStage alias from CChainState
 -:  ---------- > 17:  f0451fb98d refactor: remove CCreditPoolManager global, move to NodeContext
 -:  ---------- > 18:  a53046c4a9 refactor: remove CDSTXManager global and alias, move to CJContext
 -:  ---------- > 19:  c427305805 refactor: remove CNetFulfilledRequestManager global, move to NodeContext
 -:  ---------- > 20:  991c9ec1a2 refactor: accept NodeContext arg into BlockAssembler instead of managers
 -:  ---------- > 21:  e9de972982 refactor: move GetListAtChainTip() calls out of CGovernanceObject
 -:  ---------- > 22:  da39b73f01 refactor: move GetListAtChainTip() calls out of CGovernanceVote
 -:  ---------- > 23:  055dbba1fa refactor: move GetListAtChainTip() calls out of llmq::utils::*, misc changes
 -:  ---------- > 24:  6bd23f40aa refactor: pass CDeterministicMNManager by ref to CGovernanceManager
 -:  ---------- > 25:  d731f4127e refactor: pass CDeterministicMNManager by ref to LLMQContext members
 -:  ---------- > 26:  e628d7517a refactor: pass CDeterministicMNManager by ref to CJContext members
 -:  ---------- > 27:  d4aa891735 refactor: more passing CDeterministicMNManager by ref
 -:  ---------- > 28:  f8d1853903 refactor: even more passing CDeterministicMNManager by ref
 -:  ---------- > 29:  e432122cbd net: Move CConnman/NetEventsInterface after CNode in header file
 -:  ---------- > 30:  140e91fdca merge bitcoin#20864: Move SocketSendData lock annotation to header
 -:  ---------- > 31:  fa6847d2fb refactor: drop circular dependencies over deterministicmns in validationinterface
 1:  8ff7971286 = 32:  03d7538acf Merge #19915: p2p, refactor: Use Mutex type for some mutexes in CNode class
 2:  0c99d0a1c8 = 33:  5924442462 Merge #19469: rpc: deprecate banscore field in getpeerinfo
 3:  a965e553f9 ! 34:  ff739237b2 Merge #19464: net: remove -banscore configuration option
    @@ src/test/denialofservice_tests.cpp: BOOST_AUTO_TEST_CASE(DoS_banning)
          peerLogic->Misbehaving(dummyNode2.GetId(), 50);
          {
     @@ src/test/denialofservice_tests.cpp: BOOST_AUTO_TEST_CASE(DoS_banscore)
    -                                        m_node.llmq_ctx, false);
    +                                        ::deterministicMNManager, m_node.cj_ctx, m_node.llmq_ctx, false);
      
          banman->ClearBanned();
     -    gArgs.ForceSetArg("-banscore", "111"); // because 11 is my favorite number
 4:  8575a6b4d5 ! 35:  cc41ffe65e Merge #19512: p2p: banscore updates to gui, tests, release notes
    @@ src/test/denialofservice_tests.cpp: BOOST_AUTO_TEST_CASE(DoS_banning)
     -    auto banman = std::make_unique<BanMan>(GetDataDir() / "banlist", nullptr, DEFAULT_MISBEHAVING_BANTIME);
     -    auto connman = std::make_unique<CConnman>(0x1337, 0x1337, *m_node.addrman);
     -    auto peerLogic = PeerManager::make(chainparams, *connman, *m_node.addrman, banman.get(), *m_node.scheduler,
    --                                       *m_node.chainman, *m_node.mempool, *m_node.govman, *m_node.sporkman, m_node.cj_ctx,
    --                                       m_node.llmq_ctx, false);
    +-                                       *m_node.chainman, *m_node.mempool, *m_node.govman, *m_node.sporkman,
    +-                                       ::deterministicMNManager, m_node.cj_ctx, m_node.llmq_ctx, false);
     -
     -    banman->ClearBanned();
     -    CAddress addr1(ip(0xa0b0c001), NODE_NONE);
 5:  a39e85dc1c = 36:  6145ba1a07 Merge bitcoin/bitcoin#22495: p2p: refactor: tidy up `PeerManagerImpl::Misbehaving(...)`
```

**Comment by UdjinM6** [comment]
rebased via GH to double-check CI

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**